qiskit_experiments/library/characterization/cr_hamiltonian_analysis.py

```diff
@@ -266,21 +267,23 @@ def _generate_fit_guesses(
 
         guesses = defaultdict(list)
         for control in (0, 1):
-            # start from Z oscillation
             x_data = self._data(series_name=f"x|c={control}")
             y_data = self._data(series_name=f"y|c={control}")
             z_data = self._data(series_name=f"z|c={control}")
 
             omega_xyz = []
             for data in (x_data, y_data, z_data):
+                ymin, ymax = np.percentile(data.y, [10, 90])
+                if ymax - ymin < 0.2:
+                    # oscillation amplitude might be almost zero,
+                    # then exclude from average because of lower SNR
+                    continue
```

Comment:
What prompted this change?

Reply:
Previously, I let the FFT module estimate a frequency regardless of the signal amplitude. In some test cases, especially when the signal is really weak, the guess module picks random frequency (because computed probability has been changed), and it hurts frequency guess of averaged oscillation frequency measured in x, y, z basis. So I decided to ignore the guess when SNR is likely low, i.e. it may pick some artifact.

Reply:
That is nice idea. What is the variance here? Is it the variance of y values `np.std(data.y)` or one computed from the sampling error `data.y_err`? If latter one, probably `np.mean(data.y_err)`.